terraform/modules/hub/files/cloud-init/prometheus.sh

```diff
@@ -131,10 +131,12 @@ Environment=HTTP_PROXY=${egress_proxy_url_with_protocol}
 Environment=HTTPS_PROXY=${egress_proxy_url_with_protocol}
 EOF
 
-cat <<EOF > /etc/prometheus/prometheus.yml
-${prometheus_config}
-EOF
-
 systemctl daemon-reload
 systemctl enable  prometheus
 systemctl restart prometheus
+
+crontab -l > prometheus_config_file
```

Comment:
```
(crontab -l ;\
cat <<EOF
1 * * * * \
  aws s3 cp \
    ${config_bucket}/prometheus/prometheus.yml \
    /tmp/prometheus.yml \
  && \
  if !cmp -s /tmp/prometheus.yml /etc/prometheus/prometheus.yml; then \
    mv /tmp/prometheus.yml /etc/prometheus/prometheus.yml \
    && systemctl reload prometheus; \
  fi
EOF) | crontab -e
```

Reply:
list the crontab, append some horrible oneliner to it, write crontab from stdin

Reply:
Alternatively this could be

```
# In /etc/systemd/system/prometheus.service.d/override.conf
[Path]
PathModified=/etc/prometheus/prometheus.yml
```

and

```
(crontab -l ;\
cat <<EOF
* * * * * \
  aws s3 cp \
    ${config_bucket}/prometheus/prometheus.yml \
    /tmp/prometheus.yml \
  && \
  if !cmp -s /tmp/prometheus.yml /etc/prometheus/prometheus.yml; then \
    mv /tmp/prometheus.yml /etc/prometheus/prometheus.yml; \
  fi
EOF) | crontab -
```

Reply:
i.e. get systemd to reload prometheus using PathModified if the config changes

Reply:
We do lots of `systemctl <thing> prometheus` in here already, I'm loathe to introduce another separate file to edit to do one thing?

Reply:
Thank you! I've incorporated your changes (it's a lot nicer with some line breaks!).

Reply:
TIL that `* * * * *` is the cron syntax for "every 1 minute", not `1 * * * *` which is in fact "every hour at one minute past the hour". 🙈 🌱